scripts/pipeline_generator/plugin.py

```diff
@@ -0,0 +1,90 @@
+from pydantic import BaseModel, Field
+from typing import List, Dict, Any, Optional
+
+from .utils import HF_HOME
+
+DOCKER_PLUGIN_NAME = "docker#v5.2.0"
+KUBERNETES_PLUGIN_NAME = "kubernetes"
+
+class DockerPluginConfig(BaseModel):
+    image: str = ""
+    always_pull: bool = Field(default=True, alias="always-pull")
+    propagate_environment: bool = Field(default=True, alias="propagate-environment")
+    gpus: Optional[str] = "all"
+    mount_buildkite_agent: Optional[bool] = Field(default=False, alias="mount-buildkite-agent")
+    command: List[str] = Field(default_factory=list)
+    environment: List[str] = [
+        f"HF_HOME={HF_HOME}",
+        "VLLM_USAGE_SOURCE=ci-test",
+        "HF_TOKEN",
+        "BUILDKITE_ANALYTICS_TOKEN"
+    ]
+    volumes: List[str] = [
+        "/dev/shm:/dev/shm",
+        f"{HF_HOME}:{HF_HOME}"
+    ]
+
+class KubernetesPodContainerConfig(BaseModel):
+    image: str
+    command: List[str]
+    resources: Dict[str, Dict[str, int]]
+    volume_mounts: List[Dict[str, str]] = Field(
+        alias="volumeMounts",
+        default=[
+            {"name": "devshm", "mountPath": "/dev/shm"},
+            {"name": "hf-cache", "mountPath": HF_HOME}
+        ]
+    )
+    env: List[Dict[str, str]] = Field(
+        default=[
+            {"name": "HF_HOME", "value": HF_HOME},
+            {"name": "VLLM_USAGE_SOURCE", "value": "ci-test"},
+            {
+                "name": "HF_TOKEN",
+                "valueFrom": {
+                    "secretKeyRef": {
+                        "name": "hf-token-secret",
+                        "key": "token"
+                    }
+                }
+            },
+        ],
+    )
+
+class KubernetesPodSpec(BaseModel):
+    containers: List[KubernetesPodContainerConfig]
+    priority_class_name: str = Field(default="ci", alias="priorityClassName")
+    node_selector: Dict[str, Any] = Field(
+        default={"nvidia.com/gpu.product": "NVIDIA-A100-SXM4-80GB"},
+        alias="nodeSelector"
+    )
+    volumes: List[Dict[str, Any]] = Field(
+        default=[
+            {"name": "devshm", "emptyDir": {"medium": "Memory"}},
+            {"name": "hf-cache", "hostPath": {"path": HF_HOME, "type": "Directory"}}
+        ]
+    )
+
+class KubernetesPluginConfig(BaseModel):
+    pod_spec: KubernetesPodSpec = Field(alias="podSpec")
+
+def get_kubernetes_plugin_config(docker_image_path: str, test_bash_command: List[str], num_gpus: int) -> Dict:
```

Comment:
nit: `docker_image_path` -> `image` or `container_image` ?

k8s does not always use docker. and the image does not have to be docker image.